### DISCUSSION TITLE
"No base repo configured" error when recreating ISO

### DISCUSSION THREAD

**Discussion by luc793** [Security-Onion-Solutions/securityonion#13922] [2.4]
### Version

2.4.110

### Installation Method

Other (please provide detail below)

### Description

other (please provide detail below)

### Installation Type

other (please provide detail below)

### Location

other (please provide detail below)

### Hardware Specs

Exceeds minimum requirements

### CPU

12 (virtual)

### RAM

16 GB

### Storage for /

210 GB

### Storage for /nsm

2

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

Less than 1Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hi everyone,

I’ve been trying to generate an ISO for Security Onion, and I keep running into an issue. J

When I run the following command:

mkisofs -o securityonionCUSTOM-2.4.110-20241010.iso -allow-limited-size -b isolinux/isolinux.bin -J -R -l -c isolinux/boot.cat -no-emul-boot -boot-load-size 4 -boot-info-table -eltorito-alt-boot -e images/efiboot.img -no-emul-boot -graft-points -joliet-long -R -V "Security-Onion-2.4.110" .

I consistently get the following error in the /tmp/anaconda.log file during the kickstart install:
No base repo configured

I’ve tried recreating the default ISO without any modifications, but I’m still encountering this error. Does anyone know which command or configuration might be missing or incorrect for successfully building the ISO?

Thanks in advance for your help!

### Guidelines

- [x] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by cm-ops** [comment]
Is there a reason you aren't using the SO ISO from downloads? https://docs.securityonion.net/en/2.4/download.html

**Comment by luc793** [comment]
Thank you for your message. The actual security policies at my company require an independent and internal version of all proof of concept (POC) we are doing. As such, I need to document the entire process, including the generation of the ISO in order to complete the POC.

Could you kindly share the command (with all the necessary parameters) to generate the ISO? It would be greatly appreciated!

Best regards,

**Reply by TOoSmOotH** [reply]
So to POC Ubuntu you have to spin your own Ubuntu?

There are several projects available that can help you spin a new ISO from an existing if you need to modify the kickstart.

**Reply by luc793** [reply]
Yes I tried a variety of commansds/project in order to generate an ISO from the existing files contained in the original ISO but without success.
One of the projects is here : [https://github.com/threathunternotebook/SecurityOnion-ISO-Customization](https://github.com/threathunternotebook/SecurityOnion-ISO-Customization)
where the command used and adapted is :
`mkisofs -o securityonionCUSTOM-2.4.110-20241010.iso -allow-limited-size -b isolinux/isolinux.bin -J -R -l -c isolinux/boot.cat -no-emul-boot -boot-load-size 4 -boot-info-table -eltorito-alt-boot -e images/efiboot.img -no-emul-boot -graft-points -joliet-long -R -V "Security-Onion-2.4.110" .`
I successfully managed to boot correctly on the new ISO after creating it but I get constantly an error that the anaconda installator cannot find the sources.

**Comment by smTcos** [comment]
After spending some time searching through the documentation and various discussions, I haven’t been able to find a clear way to generate the ISO, I get the same error as you... It seems there may be a difference in the way the ISO is created (maybe an option is missing or not using correctly or perhaps, another tool used to generate).
The only useful piece of information I’ve come across is that the volume ID (using the -V option with @luc793  mkisofs command) seems to be essential for the LABEL in the kickstart config file (ks.cfg).
Maybe someone from the SO team or anyone with relevant knowledge could provide to you the exact command or steps for generating the ISO ?

**Comment by luc793** [comment]
Thank you @smTcos a lot for your message.
I continue trying various commands but still getting the same error. I tried to change the name passed through -V option, but this is affecting only the ability to initialize the kernel... I'm actually stuck further in the installation.
I hope i will get the command that SO team is using to generate the ISO :)
